armi/reactor/converters/axialExpansionChanger/assemblyAxialLinkage.py

```diff
@@ -57,118 +73,187 @@ def _determineLinked(componentA, componentB):
                 "they are going to be assumed to not be linked.",
                 single=True,
             )
-            linked = False
-        else:
-            idA, odA = (
-                componentA.getCircleInnerDiameter(cold=True),
-                componentA.getBoundingCircleOuterDiameter(cold=True),
-            )
-            idB, odB = (
-                componentB.getCircleInnerDiameter(cold=True),
-                componentB.getBoundingCircleOuterDiameter(cold=True),
-            )
+            return False
+        # Check if one component could fit within the other
+        idA = componentA.getCircleInnerDiameter(cold=True)
+        odA = componentA.getBoundingCircleOuterDiameter(cold=True)
+        idB = componentB.getCircleInnerDiameter(cold=True)
+        odB = componentB.getBoundingCircleOuterDiameter(cold=True)
+        biggerID = max(idA, idB)
+        smallerOD = min(odA, odB)
+        return biggerID < smallerOD
+    return False
 
-            biggerID = max(idA, idB)
-            smallerOD = min(odA, odB)
-            if biggerID >= smallerOD:
-                # one object fits inside the other
-                linked = False
-            else:
-                linked = True
 
-    else:
-        linked = False
+# Make a generic type so we can "template" the axial link class based on what could be above/below a thing
+Comp = typing.TypeVar("Comp", Block, Component)
```

Comment:
You'll have to teach this to me too

Reply:
This is like saying `Comp` can be used in place of `Block` or `Component`. But we can't use `typing.Union[Block, Component]` because we want `lower` and `upper` to be the same type.

If we did `Comp = typing.Union[Block, Component]`, this would be valid (from a type checker)
```python
AxialLink(lower=HexBlock(...), upper=Hexagon(...))
```
which is not what we want